sources/packages/backend/apps/workers/src/controllers/assessment/assessment.controller.ts

```diff
@@ -21,15 +25,60 @@ import {
   SupportingUserType,
 } from "@sims/sims-db";
 import { filterObjectProperties } from "../../utilities";
-import { ASSESSMENT_ID } from "../workflow-variables";
-import { ASSESSMENT_NOT_FOUND } from "apps/api/src/services";
+import {
+  ASSESSMENT_ALREADY_ASSOCIATED_TO_WORKFLOW,
+  ASSESSMENT_INVALID_OPERATION_IN_THE_CURRENT_STATE,
+  ASSESSMENT_NOT_FOUND,
+} from "../../constants";
+import {
+  ASSESSMENT_DATA,
+  ASSESSMENT_ID,
+} from "@sims/services/workflow/variables/assessment-gateway";
+import { CustomNamedError } from "@sims/utilities";
 
 @Controller()
 export class AssessmentController {
   constructor(
     private readonly studentAssessmentService: StudentAssessmentService,
   ) {}
 
+  /**
+   * Associates the workflow instance if the assessment is not associated already.
+   */
+  @ZeebeWorker("associate-workflow-instance", {
+    fetchVariable: [ASSESSMENT_ID],
+  })
+  async associateWorkflowInstance(
+    job: Readonly<
+      ZeebeJob<
+        AssociateWorkflowInstanceJobInDTO,
+        ICustomHeaders,
+        IOutputVariables
+      >
+    >,
+  ): Promise<MustReturnJobActionAcknowledgement> {
+    try {
+      await this.studentAssessmentService.associateWorkflowId(
+        job.variables.assessmentId,
+        job.processInstanceKey,
```

Comment:
is processInstanceKey available as OOTB variable that is supplied to a worker? Or is it set from our workflow code  and passed as variable here?

Reply:
The first option. It is available at the job.